client/scripts/routes.js

```diff
@@ -50,9 +57,16 @@ function config($stateProvider, $urlRouterProvider) {
       templateUrl: 'client/templates/profile.html',
       controller: 'ProfileCtrl as profile',
       resolve: {
-        user() {
-          return Meteor.user();
-        }
+        user: ['$q', function ($q) {
+          var deferred = $q.defer(),
```

Comment:
Also, is a promise that necessary here?! Why not just returning a primitive value, much simpler